includes/admin/meta-boxes/class-wc-meta-box-order-data.php

```diff
@@ -299,7 +299,16 @@ public static function output( $post ) {
 										$field_value = $order->get_meta( '_' . $field_name );
 									}
 
-									echo '<p><strong>' . esc_html( $field['label'] ) . ':</strong> ' . make_clickable( esc_html( $field_value ) ) . '</p>';
+                                    switch ( $field_name ) {
+                                        case 'billing_phone' :
+                                            $field_value = wc_make_phone_clickable( esc_html( $field_value ) );
+                                            break;
+                                        default :
+                                            $field_value = make_clickable( esc_html( $field_value ) );
+                                            break;
+                                    }
+
+                                    echo '<p><strong>' . esc_html( $field['label'] ) . ':</strong> ' . $field_value . '</p>';
```

Comment:
May as well fix this whilst you're in there; esc_html should be removed above, and then on output use `wp_kses_post` on the value.

Take care of code standards - your indenting is out.